pkg/mutate/manage_pod_mutation.go

```diff
@@ -0,0 +1,132 @@
+	switch {
+	case strategy == offv1alpha1.LocalAndRemotePodOffloadingStrategyType, strategy == offv1alpha1.RemotePodOffloadingStrategyType:
+		// The virtual-node toleration must be added.
+		toleration = getVirtualNodeToleration()
+	default:
+		// LocalPodOffloadingStrategyType, default is less clear but safer
+		// No toleration has to be added.
+		toleration = corev1.Toleration{}
+	}
+	return toleration
+}
+
+// createNodeSelectorFromNamespaceOffloading creates the right NodeSelector according to the PodOffloadingStrategy chosen.
+func createNodeSelectorFromNamespaceOffloading(nsoff *offv1alpha1.NamespaceOffloading) corev1.NodeSelector {
+	var nodeSelector corev1.NodeSelector
+	switch {
+	case nsoff.Spec.PodOffloadingStrategy == offv1alpha1.RemotePodOffloadingStrategyType:
+		// To ensure that the pod is not scheduled on local nodes is necessary to add a new NodeSelectorRequirement which
+		// requires the label "liqo.io/type=virtual-node", to every NodeSelectorTerm.
+		nodeSelector = nsoff.Spec.ClusterSelector
+		for i := range nodeSelector.NodeSelectorTerms {
+			nodeSelector.NodeSelectorTerms[i].MatchExpressions = append(nodeSelector.NodeSelectorTerms[i].MatchExpressions,
+				corev1.NodeSelectorRequirement{
+					Key:      liqoconst.TypeLabel,
+					Operator: corev1.NodeSelectorOpIn,
+					Values:   []string{liqoconst.TypeNode},
+				})
+		}
+	case nsoff.Spec.PodOffloadingStrategy == offv1alpha1.LocalAndRemotePodOffloadingStrategyType:
+		nodeSelector = nsoff.Spec.ClusterSelector
+		// A new NodeSelectorTerm terms that allows scheduling the pod also on local nodes is added.
+		newNodeSelectorTerm := corev1.NodeSelectorTerm{
+			MatchExpressions: append([]corev1.NodeSelectorRequirement{}, corev1.NodeSelectorRequirement{
+				Key:      liqoconst.TypeLabel,
+				Operator: corev1.NodeSelectorOpNotIn,
+				Values:   []string{liqoconst.TypeNode},
+			}),
+		}
+		nodeSelector.NodeSelectorTerms = append(nodeSelector.NodeSelectorTerms, newNodeSelectorTerm)
+	case nsoff.Spec.PodOffloadingStrategy == offv1alpha1.LocalPodOffloadingStrategyType:
+		// In local strategy there is no necessity to impose a particular NodeSelector.
+		nodeSelector = corev1.NodeSelector{NodeSelectorTerms: []corev1.NodeSelectorTerm{}}
+	}
+	return nodeSelector
+}
+
+// fillPodWithImposedNodeSelector gets the previously computed NodeSelector imposed by the PodOffloadingStrategy and
+// merges it with the Pod NodeSelector if it is already present. It simply adds it to the Pod if there is merge to do.
+func fillPodWithImposedNodeSelector(imposedNodeSelector *corev1.NodeSelector, pod *corev1.Pod) {
+	// To preserve the Pod Affinity content, it is necessary to add the imposedNodeSelector according to what
+	// is already present in the Pod Affinity.
+	switch {
+	case pod.Spec.Affinity == nil:
+		pod.Spec.Affinity = &corev1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{},
```

Comment:
You could directly assign the `imposedNameSelector` here, instead of creating a placeholder. The same below